src/infrastructure/api/index.ts

```diff
@@ -0,0 +1,112 @@
+import AWS from 'aws-sdk';
+import express, { Request } from 'express';
+import mysql from 'mysql2/promise';
+import vehicleQueryFunctionFactory from '../../app/vehicleQueryFunctionFactory';
+import testQueryFunctionFactory from '../../app/testQueryFunctionFactory';
+import { getResultsDetails, getVehicleDetails } from '../../domain/enquiryService';
+import ParametersError from '../../errors/ParametersError';
+import ResultsEvent from '../../interfaces/ResultsEvent';
+import VehicleEvent from '../../interfaces/VehicleEvent';
+import DatabaseService from '../databaseService';
+import SecretsManagerService from '../secretsManagerService';
+
+const app = express();
+const router = express.Router();
+
+const { API_VERSION } = process.env;
+
+// Declare middlewares
+/**
+ * bodyParser, error handling, logger, etc..
+ * http://expressjs.com/en/starter/basic-routing.html
+ * http://expressjs.com/en/guide/using-middleware.html
+ */
+
+/**
+ * app level middlewares
+ * app.use('/path', (req, res, next) => {
+ * chain middlewares
+ * next()
+ * })
+ */
+router.use((req, _response, next) => {
```

Comment:
Do we want to log the path on every request?
we may want to turn this off unless we think it adds values otherwise it can send logs to AWS that may not be useful. If I remember correctly there is a workstream going on to try to reduce the costs of logging

Reply:
Agreed